File: creational/java/singleton/src/Session.java
public class Session
{
    private final String userName;
    private final String userCode;
    private final String userType;

    private final static Session session = new Session();

    private Session()
    {
        System.out.println("Session oluşturuluyor...");
        this.userCode = "001";
        this.userName = "testUser";
        this.userType = "Admin";
        System.out.println("Session oluşturuldu...");
    }

    public static Session getSession()
    {
        return session;
    }

    public String getUserName()
    {
        return this.userName;
    }

    public String getUserCode()
    {
        return this.userCode;
    }

    public String getUserType()
    {
        return this.userType;
    }
}

File: creational/python/singleton/session.py
class session:
   __instance  = None
   __user_code = None
   __user_name = None
   __user_type = None

   @staticmethod
   def get_instance():
      if(session.__instance == None):
         session()
      return session.__instance

   def __init__(self):
      if(session.__instance != None):
         raise Exception("Sadece bir tane session nesnesi yaratılabilir.")
      else:
         self.__user_type   = "Admin"
         self.__user_name   = "testUser"
         self.__user_code   = "001"
         session.__instance = self

   def get_user_name(self):
      return self.__user_name

   def set_user_name(self, new_name):
      self.__user_name = new_name

   def get_user_code(self):
      return self.__user_code

   def set_user_code(self, new_code):
      self.__user_code = new_code

   def get_user_type(self):
      return self.__user_type

   def set_user_type(self, new_type):
      self.__user_type = new_type

session = session().get_instance()

print(session.get_user_name(), session.get_user_code(), session.get_user_type())
